File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        res = ''
        resLength = 0
        
        for i in range(len(s)):
        
          # Odd lengths
          start, end = i, i
          while start >= 0 and end < len(s) and s[start] == s[end]:
            if end - start + 1 > resLength:
              res = s[start:end+1]
              resLength = end - start + 1
            start -= 1
            end += 1
          
          # Even lengths
          start, end = i, i + 1
          while start >= 0 and end < len(s) and s[start] == s[end] :
            if end - start + 1 > resLength:
              res = s[start:end+1]
              resLength = end - start + 1
            start -= 1
            end += 1
          
        return res
    

    
   

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.js
/**
 * @param {string} s
 * @return {string}
 */
var longestPalindrome = function(s) {  
  let result = "";
  let resLength = 0;
  
  for (let i = 0; i < s.length; i++) {

    // Check for odd-length 
    let start = i, end = i;
    while (s[start] === s[end] && start >= 0 && end < s.length) {
      if (end - start + 1 > resLength) {
        result = s.substring(start, end + 1);
        resLength = end - start + 1;
      }
      start--;
      end++;
    }
    
    // Check for even-length
    start = i, end = i + 1;
    while (s[start] === s[end] && start >= 0 && end < s.length) {
      if (end - start + 1 > resLength) {
        result = s.substring(start, end + 1);
        resLength = end - start + 1;
      }
      start--;
      end++;
    }
  }
 
  return result;
};

